internal/procedures/server/start.go

```diff
@@ -47,15 +47,49 @@ func Start(initiator common.ServerInitiator) (err error) {
 		go startCourse(course)
 	}
 
-	// Cleanup any temp dirs.
-	defer util.RemoveRecordedTempDirs()
+	return nil
+}
 
-	err = server.RunServer(initiator)
-	if err != nil {
-		return fmt.Errorf("Error during server startup sequence: '%w'.", err)
+func CleanupAndStopServer() (err error) {
+	if serverInstance == nil {
+		return nil
 	}
 
-	log.Debug("Server closed.")
+	err = errors.Join(err, db.Close())
+	err = errors.Join(err, util.RemoveRecordedTempDirs())
+
+	serverInstance.apiServer.StopServer()
+
+	serverInstance.apiServer = nil
+	serverInstance = nil
+
+	return err
+}
+
+func RunAndBlockServer(initiator common.ServerInitiator, skipSetup bool) (err error) {
```

Comment:
Given the name of this package, I would drop "Server" from these function names.